OSDOCS 9556: Release note udate about the default channel in mirror operator catalog packages (#71528)
openshift/openshift-docs: release_notes/ocp-4-15-release-notes.adoc

[id="ocp-4-15-openshift-cli-bug-fixes"]

> ==== OpenShift CLI (oc)

Reviewer: OCPBUGS-385 is for oc-mirror plugin. It doesn't relate to oc. I'm not sure if we should create a sub-section oc-mirror under here? But we cannot leave it directly under oc in my opinion.

Author: How about I add the updates in Image registry?

> Prior to this release, `mirror.operators.catalog.packages` section did not define the channel the user want to use.

Author: @sherine-k can you ptal?

> Prior to this release, `mirror.operators.catalog.packages` section did not define the channel the user want to use.

Reviewer: ```suggestion
Prior to this release, when filtering operator packages by channel (`mirror.operators.catalog.packages.channels`), the user had to add the default channel for that package, even if he/she is not interested in using bundles from that channel: the catalog was considered invalid if it defined a default channel that is not included in the catalog.
```

> Prior to this release, `mirror.operators.catalog.packages` section did not define the channel the user want to use.

Author: Ack.

+

> With this update, the field `defaultChannel`is introduced in the section for `mirror.operators.catalog.packages` that allows the user to define what channel they want to define as the `default channel`.

Reviewer: ```suggestion
With this update, the field `defaultChannel` is introduced in the section for `mirror.operators.catalog.packages`. It allows the user to select a channel of his choosing as the default. Based on this information, oc-mirror will build a new catalog, defining the channel selected in `defaultChannel` as the package's default. 
```

Author: ack.

> With this update, the field `defaultChannel`is introduced in the section for `mirror.operators.catalog.packages` that allows the user to define the channel they want to use instead of the set defaultChannel in the operator, this acts as a defaultChannel override.

Author: ack...

[id="ocp-4-15-openshift-cli-bug-fixes"]
==== OpenShift CLI (oc)

[discrete]
[id="ocp-4-15-openshift-oc-mirror-bug-fixes"]
==== oc-mirror

> Prior to this release, if a user wanted to filter operator packages by channel, that is, `mirror.operators.catalog.packages.channels`, they were required to include the default channel for the package, even if they had no intention of using bundles from that channel.  Based on this information, if the package had a default channel that was not part of the catalog, the catalog is considered invalid.

Reviewer: ```suggestion
Prior to this release, if a user wanted to filter operator packages by channel, that is, `mirror.operators.catalog.packages.channels`, they were required to include the default channel for the package in the imageSetConfig, even if they had no intention of using bundles from that channel.  This increased the volume of mirrored images, but kept the default channel intact in the catalog, which guarantied better upgrades for the operator.
```
the default channel is in the catalog, we' re asking the user to include it in the imagesetconfig. I think this is the confusing part. 
So we cannot use the statement :
```
Based on this information, if the package had a default channel that was not part of the catalog, the catalog is considered invalid.
```
Or we should say:
```
Based on this information, if the imageSetConfig did not include the default channel for the package,  the resulting catalog is considered invalid.
```

Author: Ack.

> Prior to this release, if you wanted to filter operator packages by channel, that is, `mirror.operators.catalog.packages.channels`, you had to specify the default channel for the package, even if you did not intend to use the packages from that channel. Based on this information, the resulting catalog is considered invalid if the `imageSetConfig` does not contain the default channel for the package.

Author: ack.

> Prior to this release, if a user wanted to filter operator packages by channel, that is, `mirror.operators.catalog.packages.channels`, they were required to include the default channel for the package, even if they had no intention of using bundles from that channel.  Based on this information, if the `imageSetConfig` did not include the default channel for the package,  the resulting catalog is considered invalid.

Reviewer 2: ```suggestion
Prior to this release, if you wanted to filter operator packages by channel, that is, `mirror.operators.catalog.packages.channels`, you had to specify the default channel for the package, even if you did not intend to use the packages from that channel. Based on this information, the resulting catalog is considered invalid if the `imageSetConfig` does not contain the default channel for the package.
```

Author: Ack.

[WARNING]
====

> Modifying the default channel is very risky. Be aware that all the responsibility falls on the user. 

Reviewer: ```suggestion
Modifying the default channel should not be taken lightly as it might have an impact on the upgrade paths of the operator. Using `defaultChannel` is under the responsibility of the end user.. 
```

Author: Ack.

> Modifying the default channel should not be taken lightly as it could have an impact on the operator's upgrade paths. The use of `defaultChannel` is the responsibility of the end user.

Author: Ack.

> Modifying the default channel should not be taken lightly as it might have an impact on the upgrade paths of the operator. Using `defaultChannel` is under the responsibility of the end user.

Reviewer 2: ```suggestion
Modifying the default channel should not be taken lightly as it could have an impact on the operator's upgrade paths. The use of `defaultChannel` is the responsibility of the end user.
```

Author: Ack

====

> This update introduces the `defaultChannel` field in the `mirror.operators.catalog.packages` section. You can now select a default `channel`. This action enables `oc-mirror` to build a new catalog that defines the selected channel in the `defaultChannel` field as the default for the package. 

Author: ack.

> With this update, the field `defaultChannel` is introduced in the section for `mirror.operators.catalog.packages`. It allows the user to select a default `channel`. Based on this information, oc-mirror builds a new catalog, defining the selected channel in `defaultChannel` field as the package's default. 

Reviewer 2: ```suggestion
This update introduces the `defaultChannel` field in the `mirror.operators.catalog.packages` section. You can now select a default `channel`. This action enables `oc-mirror` to build a new catalog that defines the selected channel in the `defaultChannel` field as the default for the package.
```

Author: Ack

> For example, if `operator A` has a catalog with four channels, and the user is interested in `stable-4.14` instead of the default channel. You can add the desired default channel manually in the given field. oc-mirror now builds a new catalog with the mentioned default channel, overriding the previous default channel. 

Reviewer: ```suggestion
For example, a catalog contains `operator A` with several channels, and `stable-4.15` as the default channel. The user is interested only in `stable-4.14` instead of the default channel. You can add the desired default channel to the imageSetConfig under the package's `defaultChannel`. oc-mirror now builds a new catalog with the defined default channel, overriding the previous default channel. 
```

Author: Ack.

> For example, a catalog contains `operator A` with several channels, and `stable-4.15` as the default channel. The user is interested only in `stable-4.14` instead of the default channel. You can add the desired default channel to the `imageSetConfig` under the package's `defaultChannel`. oc-mirror now builds a new catalog with the defined default channel, overriding the previous default channel. 

Reviewer 2: ```suggestion
If you are interested in a specific channel, such as `stable-4.14`, and not the default channel, you can add the desired default channel to the `imageSetConfig` under the `defaultChannel` of the package. This action enables `oc-mirror` to build a new catalog with the defined default channel and thus override the previous default channel, for example, in a catalog containing `operator A` with multiple channels and `stable-4.15` as the default channel.
```

Author: ack.

> If you are interested in a specific channel, such as `stable-4.14`, and not the default channel, you can add the desired default channel to the `imageSetConfig` under the `defaultChannel` of the package. This action enables `oc-mirror` to build a new catalog with the defined default channel and thus override the previous default channel, for example, in a catalog containing `operator A` with multiple channels and `stable-4.15` as the default channel. 

Author: Okay..Ack

> * Before introducing the `defaultChannel` field

Reviewer 2: ```suggestion
.Before introducing the `defaultChannel` field
```

Author: ack.

[source,yaml]
----
mirror:
  operators:
  - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.15
    packages:
    - name: operatorA
      channels:
      - name: stable-4.14 # what the user needs
      - name: stable-4.15 # needed as it is default channel, to avoid error
----

> * After introducing the `defaultChannel` field

Reviewer 2: ```suggestion
.After introducing the `defaultChannel` field
```

Author: ack.

[source,yaml]
----
mirror:
  operators:
  - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.15
    packages:
    - name: operatorA
      defaultChannel: stable-4.14
      channels:
      - name: stable-4.14 # what the user needs
----

> An initial check is added to ensure that if the `defaultChannel` field is set that the user must also define it in the channels section of the `ImageSetConfig`.

Reviewer 2: ```suggestion
An initial check is added to ensure that if the `defaultChannel` field is set, you must also define it in the channels section of `ImageSetConfig`.
```

> An initial check is added to ensure that if the `defaultChannel` field is set, you must also define it in the channels section of `ImageSetConfig`.

Author: Ack.

[id="ocp-4-15-openshift-cli-bug-fixes"]
==== OpenShift CLI (oc)

[discrete]
[id="ocp-4-15-openshift-oc-mirror-bug-fixes"]
==== oc-mirror

Prior to this release, to filter operator packages by channel, for example, `mirror.operators.catalog.packages.channels`, you had to specify the default channel for the package, even if you did not intend to use the packages from that channel. Based on this information, the resulting catalog is considered invalid if the `imageSetConfig` does not contain the default channel for the package.

[WARNING]
====

> Modifying the default channel should not be taken lightly as it could have an impact on the Operator's upgrade paths. The use of `defaultChannel` is the responsibility of the end user.

Reviewer 3: `The use of defaultChannel is the responsibility of the end user.` This can be replaced by when the user must be using.

```suggestion
Modifying the default channel can have an impact on the Operator's upgrade paths. The use of `defaultChannel` is the responsibility of the end user.
```

Author: ack.

====

This update introduces the `defaultChannel` field in the `mirror.operators.catalog.packages` section. You can now select a default channel. This action enables `oc-mirror` to build a new catalog that defines the selected channel in the `defaultChannel` field as the default for the package. 

> If you are interested in a specific channel, such as `stable-4.14`, and not the default channel, you can add the required default channel to the `imageSetConfig` under the `defaultChannel` of the package. This action enables `oc-mirror` to build a new catalog with the defined default channel and override the previous default channel, for example, in a catalog containing `operator A` with multiple channels and `stable-4.15` as the default channel.

Reviewer 3: ```suggestion
If you want to use a specific channel, such as `stable-4.14`, and not the default channel, you can add the required default channel to the `imageSetConfig` under the `defaultChannel` of the package. This action enables `oc-mirror` to build a new catalog with the defined default channel and override the previous default channel.
```

Author: Ack.

> If you are interested in a specific channel, such as `stable-4.14`, and not the default channel, you can add the required default channel to the `imageSetConfig` under the `defaultChannel` of the package. This action enables `oc-mirror` to build a new catalog with the defined default channel and override the previous default channel, for example, in a catalog containing `operator A` with multiple channels and `stable-4.15` as the default channel.

Reviewer 3: `This action enables oc-mirror to build a new catalog with the defined default channel and override the previous default channel`
This is repetitive. Prev sentence has the same.

Author: Let me remove it.

.Before introducing the `defaultChannel` field
[source,yaml]
----
mirror:
  operators:
  - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.15
    packages:
    - name: operatorA
      channels: 
      - name: stable-4.14 # what the user needs
      - name: stable-4.15 # needed as it is default channel, to avoid error
----

.After introducing the `defaultChannel` field
[source,yaml]
----
mirror:
  operators:
  - catalog: registry.redhat.io/redhat/redhat-operator-index:v4.15
    packages:
    - name: operatorA
      defaultChannel: stable-4.14 <1>
      channels:
      - name: stable-4.14 # what the user needs
----

> <1> Specify the required default channel.

Reviewer 3: This needs to be documented in oc-mirror docs and not in release notes. Please add this in the oc-mirror docs and provide a reference to the docs here.

Author: Ack.

> An initial check is added to ensure that if the `defaultChannel` field is set, you must also define it in the `channels` section of `ImageSetConfig`

Reviewer 3: This will also be a part of oc-mirror docs.

Author: Ack.
